### PR TITLE
Handle new import path for private Keras modules in 2.13

### PR DESCRIPTION
**Pull request by ianstenbit**
With 2.13, we can't import these private API symbols directly but instead have to use `keras.src`.

This change keeps support for 2.12 while also enabling the `keras.src` path for 2.13

### CODE REVIEW

**Review by haifeng-jin — APPROVED** [review]
Thanks for the PR! LGTM.

**Comment by haifeng-jin** [comment]
Seems the gcbrun has failed.

File: keras_cv/models/object_detection/predict_utils.py [hunk 1]
```diff
@@ -12,10 +12,17 @@
...
-from keras.utils import tf_utils
+
+if hasattr(keras, "src"):
+    from keras.src.engine.training import _minimum_control_deps
+    from keras.src.engine.training import reduce_per_replica
+    from keras.src.utils import tf_utils
+else:
+    from keras.engine.training import _minimum_control_deps
+    from keras.engine.training import reduce_per_replica
+    from keras.utils import tf_utils
```
[the remark this answers is not in the document]
Comment (ianstenbit) [reply keras_cv/models/object_detection/predict_utils.py]: Good call -- I've updated to use this approach

**Comment by ianstenbit** [comment]
> Seems the gcbrun has failed.

This looks like it was probably a flake so I'm going to re-run

**Pull request merged**